lib/rmg/CRmgTemplateZone.cpp

```diff
@@ -2243,41 +2247,48 @@ void CRmgTemplateZone::addAllPossibleObjects()
 	oi.maxPerZone = std::numeric_limits<ui32>().max();
 
 	//dwellings
+	auto dwellingTypes = {Obj::CREATURE_GENERATOR1, Obj::CREATURE_GENERATOR4};
 
-	auto subObjects = VLC->objtypeh->knownSubObjects(Obj::CREATURE_GENERATOR1);
-
-	//don't spawn original "neutral" dwellings that got replaced by Conflux dwellings in AB
-	static int elementalConfluxROE[] = { 7, 13, 16, 47 };
-	for (int i = 0; i < 4; i++)
-		vstd::erase_if_present(subObjects, elementalConfluxROE[i]);
-
-	for (auto secondaryID : subObjects)
+	for(auto dwellingType : dwellingTypes)
 	{
-		auto dwellingHandler = dynamic_cast<const CDwellingInstanceConstructor*>(VLC->objtypeh->getHandlerFor(Obj::CREATURE_GENERATOR1, secondaryID).get());
-		auto creatures = dwellingHandler->getProducedCreatures();
-		if (creatures.empty())
-			continue;
+		auto subObjects = VLC->objtypeh->knownSubObjects(dwellingType);
 
-		auto cre = creatures.front();
-		if (cre->faction == townType)
+		if(dwellingType == Obj::CREATURE_GENERATOR1)
```

Comment:
What do you think about fixing or creating a wrapper over VLC->objtypeh->knownSubObjects to consider if needed ConfluxROE or not? Thus, the code like this vstd::erase_if_present(subObjects, elementalConfluxROE[i]); will be not needed.

Reply:
I think h3m map can contain old dwellings, doesn't it? So we still need to be able to construct them but we do not want to put them on the random map?

Reply:
We probably can add a configuration to json like obsolete: true

Reply:
This is pure RMG logic not needed for any other place. Might be good to introduce some rmg.blocked=true which will block using this object by RMG in json config but this is a complex task and I do not want to do it now.